### ISSUE TITLE
Lack of explanation of difference between data descriping models

### ISSUE THREAD

**Issue by Make42** [allegroai/clearml-docs#477]
It is unclear what the differences are conceptually between the following things.

In the web app there are [multiple types of descriptive data available for models](https://clear.ml/docs/latest/docs/webapp/webapp_model_viewing/#model-configuration)

- tag
- model configuration (in the tab "network")
- metadata

Those can also be set and gotten programmatically.

Conceptually they are all descriptive data, but it is not clear to me what the differences conceptually are. Here are some ideas of mine:

- Model configuration contains values that are available before the training, e.g., hyperparameters, paths where the model shall be saved to, etc.
- metadata contains values that are available after training, e.g., how many training iterations where done, how long training took, how often a certain function was called, performance values (like accuracy)
- tags... no idea... the documentation exemplified "Keras" and "example". So maybe these are supposed to be values that are available before the training and that describe the model, but that cannot be changed without rewriting the entire code, in contrast to hyperparameters?

**Question 1:** How are they conceptually different?

Maybe they are not supposed to be conceptually different, but just technologically. For example, you have the two methods `connect`  and `connect_configuration` of `Task`, which do not differ in the conceptual content they contain, but in that the first connects Python symbols/variables to the task and the second connects files to the task.

**Question 2:** How are they technologically different?

In practice, the reason why I populate those data fields before/during/after training is so that I can use them as query arguments, when looking for appropriate models when I want to use one for inference later. One of the most important query arguments here is the hyperparameters. Assuming my understanding above was correct, I do not see how I can include the model configuration into my query...!

**Comment by Make42** [comment]
If someone explains those things here, I am happy to make a pull request.
